docs/library/axi_fan_control/index.rst

```diff
@@ -0,0 +1,265 @@
+   * - TEMP_00_L
+     - Temperature threshold in degree celcius above which PWM should be 100%.
+
+Interface
+--------------------------------------------------------------------------------
+
+.. hdl-interfaces::
+
+   * - temp_in
+     - Input bus for use with System Management Wizzard IP.
+   * - tacho
+     - Tacho generator input.
+   * - pwm
+     - PWM control signal.
+   * - irq
+     - Interrupt signal, level high.
+   * - s_axi
+     - AXI Slave Memory Map interface.
+
+Clocking
+--------------------------------------------------------------------------------
+
+The IP core runs on the AXI clock and requires a frequency of 100MHz.
+
+Detailed Description
+--------------------------------------------------------------------------------
+
+The main features of this IP core are its independent operation and the fact
+that it does not require an external temperature sensor. All of the mechanisms
+contained inside the core are controlled by a state machine, so that they do not
+depend on the software in case the software fails. The state machine uses the
+temperature it reads from the SYSMONE4 primitive or via the "temp_in" bus to
+decide the correct PWM duty-cycle. The temperature thresholds and hysteresis
+have defaults set in hardware and can be modified by the software. The
+INTERNAL_SYSMONE parameter is used to set the temperature values source, 0 when
+reading from temp_in and 1 when instantiating the internal SYSMONE primitive.
+
+Running independently
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The hardware can operate with no input from the software; the IP core starts
+working after the bitstream is loaded, without the need to be brought out of
+reset. In order to activate the interrupts the software must write to the
+IRQ_MASK register. At this point the hardware starts operating and a minimal
+feedback is provided.
+
+There are 9 temperature intervals defined in the hardware as below:
+
+.. image:: pwm_vs_temp.svg
+   :alt: PWM vs Temperature
+
+Five of these intervals have only one possible duty-cycle and four of them can
+have either of the neighbouring values. After reset the PWM duty-cycle will
+start as 100%. The state-machine will begin reading the temperature and will
+decide on the PWM duty cycle depending on which interval the value matches. The
+PWM duty-cycle will only change when the temperature enters one of the five
+intervals with a single PWM duty-cycle, in the other four the previous
+duty-cycle will be maintained. In these intervals its value will depend on
+whether the temperature is rising or falling. The temperature can be
+reconfigured by the software.
+
+The temperature is obtained from the PL SYSMONE4 primitive as a 16 bit raw value
+or from the temp_in bus as 10 bit. This value can also be accessed by the
+software using the TEMPERATURE register however the reading is done periodically
+and overwrites the register so only the most value will be available. In order
```

Comment:
"The most **recent** value"?

Reply:
Yes, but rephrased as :
```
The latest reading is in the TEMPERATURE register.
```
No need to explain how registers work ._.

Reply:
Yep, much better. Don't know why I haven't thought of it 👍🏼